### DISCUSSION TITLE
z.date() does not work with vi.setSystemTime()

### DISCUSSION THREAD

**Discussion by Its-treason** [colinhacks/zod#2764] [Q&A] [answered]
When mocking the system, time with vitest (which uses: `https://github.com/sinonjs/fake-timers` internally) the date parser stops working.

CodeSandbox: https://codesandbox.io/p/sandbox/zen-bartik-gg4nn9

```ts
it("runs", () => {
  const dateSchema = z.coerce.date();

  // This Works
  console.log(dateSchema.parse("2023-07-19"));

  vi.setSystemTime(1689775661);

  // This doesn't work
  console.log(dateSchema.parse("2023-07-19"));
});
```

The second parse throws an Error, even through it worked before with the same value:
```
ZodError: [
  {
    "code": "invalid_type",
    "expected": "date",
    "received": "object",
    "path": [],
    "message": "Expected date, received object"
  }
]
```

**Comment by JacobWeisenburger** [comment, marked as answer]
This seems to have nothing to do with Zod.

https://vitest.dev/api/vi.html#vi-setsystemtime

you need to use `vi.useFakeTimers()`



If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by agos** [reply]
fyi the example links are broken
